arviz/stats/stats.py

```diff
@@ -1262,10 +1269,11 @@ def summary(
         )
 
     # Combine metrics
-    metrics = []
-    metric_names = []
+    metrics: List[xr.Dataset] = []
+    metric_names: List[str] = []
     if extend:
-        metrics_names_ = (
+        metrics_: Tuple[xr.Dataset, ...]
+        metrics_names_: Tuple[str, ...] = (
```

Comment:
The block below is why I noticed that the `kind` error check was missing, and added it: if `kind` is not one of `"all", "stats", "diagnostics"`, then `metrics_names_` will remain set to the tuple of strings, but `metrics_` will never get assigned — even though it's about to get used at the end of the `if-elif-elif` block. This would have led to a user-unfriendly `UnboundLocalError`.

With the error check I added, supplying an illegal `kind` value is treated the same way as supplying an illegal `order` or `fmt` value: with a `TypeError` containing a descriptive and actionable error message.